lib/compress.js

```diff
@@ -96,6 +96,8 @@ function Compressor(options, false_by_default) {
         unsafe_undefined: false,
         unused          : !false_by_default,
     }, true);
+    var evaluate = this.options["evaluate"];
+    this.eval_threshold = /eager/.test(evaluate) ? 1 / 0 : +evaluate;
```

Comment:
Could this `this.options["evaluate"]` ever be `undefined`?  If so, `eval_threshold` would be `NaN`:
```
$ node
> var evaluate = void 0;  /eager/.test(evaluate) ? 1 / 0 : +evaluate;
NaN
> var evaluate = null; /eager/.test(evaluate) ? 1 / 0 : +evaluate;
0
> var evaluate = 0; /eager/.test(evaluate) ? 1 / 0 : +evaluate;
0
> var evaluate = false; /eager/.test(evaluate) ? 1 / 0 : +evaluate;
0
> var evaluate = true; /eager/.test(evaluate) ? 1 / 0 : +evaluate;
1
> var evaluate = "123"; /eager/.test(evaluate) ? 1 / 0 : +evaluate;
123
```

Reply:
Only if you explicitly pass in `undefined`, `NaN` or some weird strings, otherwise by default it will be `!false_by_default`.

And since `n < NaN` is `false` all the way, so all it does is disabling `evaluate` which seems fine to me 😎

Reply:
Okay. I forgot about `!false_by_default`. It works as it did before. 👍 
```
$ echo 'console.log(+1);' | bin/uglifyjs -c evaluate=undefined
console.log(1);
$ echo 'console.log(+1);' | bin/uglifyjs -c evaluate=null
console.log(+1);
$ echo 'console.log(+1);' | bin/uglifyjs -c evaluate=false
console.log(+1);
$ echo 'console.log(+1);' | bin/uglifyjs -c evaluate=0
console.log(+1);
```